src/raphael_backend_flask/llm.py

```diff
@@ -1,36 +1,41 @@
 import json
 from typing import Any, Iterable
 
-from health_misinfo_shared.fine_tuning import infer_claims
+from health_misinfo_shared.fine_tuning import (
+    infer_transcript_claims,
+    infer_multimodal_claims,
+)
+from raphael_backend_flask import multimodal
 from raphael_backend_flask.db import (
     execute_sql,
     update_claim_extraction_run,
 )
 from health_misinfo_shared.label_scoring import get_claim_summary
-from raphael_backend_flask.process import refine_offsets
+from raphael_backend_flask.transcript import refine_offsets
 
 
-def extract_claims(run: dict) -> Iterable[dict[str, Any]]:
+def extract_transcript_claims(run: dict) -> Iterable[dict[str, Any]]:
     sentences = run["transcript"]
-    inferred_claims = infer_claims(run["id"], sentences)
+    inferred_claims = infer_transcript_claims(sentences)
 
     for response in inferred_claims:
-        claims = response.get("response")
+        claims = response.get("response", {})
         chunk = response.get("chunk")
+        if not chunk:
+            raise Exception("Could not extract claims: got no chunks")
+
         for claim in claims:
             labels_dict = claim.get("labels", {})
             labels_dict["summary"] = get_claim_summary(labels_dict)
-            # checkworthiness = claim.get("labels", {}).get("summary", "na")
-            # checkworthiness will be one of "worth checking", "may be worth checking" or "not worth checking"
             parsed_claim = {
                 "run_id": run["id"],
-                "claim": claim["claim"],
-                "raw_sentence_text": claim["original_text"],
+                "claim": claim.get("claim", "Probably a bug - please report"),
+                "raw_sentence_text": claim.get("original_text", ""),
                 "labels": json.dumps(labels_dict),
-                "offset_start_s": chunk["start_offset"],
+                "offset_start_s": chunk.get("start_offset", 0),
             }
-            if chunk["end_offset"] is not None:
-                parsed_claim["offset_end_s"] = chunk["end_offset"]
+            if end := chunk.get("end_offset") is not None:
+                parsed_claim["offset_end_s"] = end
```

Comment:
Bug here – this sets `end` to a boolean, rather than a float.

Reply:
Very good spot.

Reply:
```python
if (end := chunk.get("end_offset")) is not None:
```
would probably be correct, but given that it's super easy to miss why the brackets are there, I've opted to remove the walrus.